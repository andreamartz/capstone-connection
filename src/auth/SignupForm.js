import React, { useState, useRef } from 'react';
import { useHistory } from "react-router-dom";
import CapConApi from "../api/api";
// import Alert from "../common/Alert";
import Alert from '@material-ui/lab/Alert';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import AddCircleOutlineOutlinedIcon from '@material-ui/icons/Add';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
// import './SignupForm.css';

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(16),
    marginLeft: 'auto',
    marginRight: 'auto',
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
    paddingLeft: theme.spacing(4),
    paddingRight: theme.spacing(4),
    display: 'flex',
    flexDirection: 'column',
    height: 'auto',
    width: '40rem',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.warning.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  }
}));


const SignupForm = ({ signup }) => {
  const [formData, setFormData] = useState({
    username: "",
    password: "",
    firstName: "",
    lastName: "",
    email: "",
    bio: "",
    photoUrl: "",
    portfolioUrl: "",
    gitHubUrl: ""
  });
  const [formErrors, setFormErrors] = useState([]);
  // const [fileInputState, setFileInputState] = useState('');

  const classes = useStyles();
  const history = useHistory();

  console.debug(
    "SignupForm",
    "formData=", formData,
    "formErrors", formErrors
  );

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    let result = await signup(formData);
    if (result.success) {
      history.push("/projects");
    } else {
      setFormErrors(result.errors);
    }
  }

  /** Update form data field */
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
  }

  // **************
  // Get handleFileInputChange(e), previewFile(file), handleSubmitFileetc. for image upload from NewPrj after bugs are worked out
  // *************
  // const handleFileInputChange = (e) => {
  //   const file = e.target.files[0];
  //   setFormData(data => ({ ...data, photoUrl: file }));
    
  //   previewFile(file);
  // };

  // const previewFile = (file) => {
  //   const reader = new FileReader();
  //   reader.readAsDataURL(file);  
  //   reader.onloadend = () => {
  //     setPreviewSource(reader.result);
  //   };
  // };

  // const handleSubmitFile = async (e) => {
  //   console.log('submitting');
  //   e.preventDefault();
  //   if (!previewSource) return;
  //   console.log("PREVIEWSOURCE: ", previewSource);
  //   let result = await CapConApi.addProject(formData);
  //   if (result.success) {
  //     history.push("/projects");
  //   } else {
  //     setFormErrors(result.errors);
  //   }

  //   // uploadImage(previewSource);
  // };

  return (
    <Paper className={classes.paper} elevation={5} component="main" maxWidth="xs">
      <Avatar className={classes.avatar}>
        <AddCircleOutlineOutlinedIcon />
      </Avatar>
      <Typography component="h1" variant="h5">
        Signup
      </Typography>
      {/* <form onSubmit={handleSubmit} className={classes.form} noValidate> */}
      <form className={classes.form} noValidate>
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="username"
          label="Username"
          name="username"
          onChange={handleChange}
          type="text"
          value={formData.username}
          autoComplete="username"
          autoFocus
        />
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="password"
          label="Password"
          name="password"
          type="password"
        />
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="firstName"
          label="First name"
          name="firstName"
          type="firstName"
        />
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="lastName"
          label="Last name"
          name="lastName"
          type="lastName"
        />
        <TextField
          variant="outlined"
          margin="normal"
          fullWidth
          id="email"
          label="Email address"
          name="email"
          type="email"
        />
        <TextField
          variant="outlined"
          margin="normal"
          fullWidth
          id="bio"
          label="Bio"
          name="bio"
          type="bio"
        />
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="photoUrl"
          label="URL of avatar"
          name="photoUrl"
          type="photoUrl"
        />
        <TextField
          variant="outlined"
          margin="normal"
          fullWidth
          id="portfolioUrl"
          label="URL of portfolio site"
          name="portfolioUrl"
          type="portfolioUrl"
        />
        <TextField
          variant="outlined"
          margin="normal"
          fullWidth
          id="gitHubUrl"
          label="URL of GitHub profile"
          name="gitHubUrl"
          type="gitHubUrl"
        />

        {/* {formErrors.length
          ? <Alert severity="error">This is an error alert â€” check it out!</Alert>
          : null
        } */}
        <Button
          type="submit"
          fullWidth
          variant="contained"
          color="primary"
          className={classes.submit}
          size="large"
        >
          Sign In
        </Button>
      </form>
    </Paper>
  );
}

export default SignupForm;