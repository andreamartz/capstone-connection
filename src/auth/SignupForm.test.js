import React from 'react';
import { render } from '@testing-library/react';
import SignupForm from './SignupForm';
import { MemoryRouter } from 'react-router';

describe('Smoke test', () => {
  test('it renders without crashing', () => {
    render(<SignupForm />);
  });
});
