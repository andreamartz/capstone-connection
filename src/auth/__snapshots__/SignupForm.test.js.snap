// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test it matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiPaper-root makeStyles-paper-17 MuiPaper-elevation5 MuiPaper-rounded"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-18 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5"
    >
      Signup
    </h1>
    <form
      class="makeStyles-form-20"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="username"
          id="username-label"
        >
          Username
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="username"
            name="username"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Username
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="firstName"
          id="firstName-label"
        >
          First name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="firstName"
            name="firstName"
            type="firstName"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                First name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="lastName"
          id="lastName-label"
        >
          Last name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="lastName"
            name="lastName"
            type="lastName"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Last name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="email"
          id="email-label"
        >
          Email address
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="email"
            name="email"
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Email address
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="bio"
          id="bio-label"
        >
          Bio
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="bio"
            name="bio"
            type="bio"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                Bio
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-28"
      >
        <div
          class="MuiBox-root MuiBox-root-29"
        >
          <div
            class="MuiBox-root MuiBox-root-30"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Photo or avatar image
              <sup>
                *
              </sup>
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Choose File
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiBox-root MuiBox-root-31 makeStyles-photoUrlContainer-22"
          >
            <input
              class="makeStyles-photoUrl-21"
              id="photoUrl"
              label="URL of avatar"
              margin="normal"
              name="photoUrl"
              type="file"
              value=""
              variant="outlined"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-32"
      />
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="portfolioUrl"
          id="portfolioUrl-label"
        >
          URL of portfolio site
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="portfolioUrl"
            name="portfolioUrl"
            type="portfolioUrl"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                URL of portfolio site
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="gitHubUrl"
          id="gitHubUrl-label"
        >
          URL of GitHub profile
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="gitHubUrl"
            name="gitHubUrl"
            type="gitHubUrl"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-24 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-26"
            >
              <span>
                URL of GitHub profile
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-23 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Sign Up
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </main>
</DocumentFragment>
`;
